server/analytics/analytics.go

```diff
@@ -52,24 +52,23 @@ func OpenDB(dsn string) (*Analytics, error) {
 		return nil, err
 	}
 
-	var clusterArgs ClusterArgs
-	isCluster := false
-
-	if strings.Contains(dsn, "&cluster") && strings.Contains(dsn, "&local_suffix") {
-		isCluster = true
-		dsnArgs := strings.Split(dsn, "&")
+	var clusterArgs *ClusterArgs
+	parsedDSN, err := url.Parse(dsn)
+	if err != nil {
+		return nil, err
+	}
 
-		for _, v := range dsnArgs {
-			splitted := strings.Split(v, "=")
-			key := splitted[0]
-			value := splitted[1]
-			if key == "cluster" {
-				clusterArgs.Cluster = value
-			} else if key == "local_suffix" {
-				clusterArgs.LocalSuffix = value
-			}
-		}
+	if parsedDSN.Query().Has("cluster") && parsedDSN.Query().Has("local_suffix") {
+		clusterArgs.ClusterName = parsedDSN.Query().Get("cluster")
```

Comment:
One little nitpicking: If the field name is called `clusterName` I would suggest to name the query param accordingly (`clusterName`).